tests/foreman/rhci/test_computeresources.py

```diff
@@ -0,0 +1,353 @@
+    def test_create_compute_resource(self, data):
+        """ @Test: Create a compute resource.
+        @Assert: A compute resource is created
+        @Feature: Compute Resource
+        """
+        with Session(self.browser) as session:
+            make_resource(
+                session,
+                name=data['name'],
+                provider_type=data['provider'],
+                parameter_list=[
+                    [data['url_name'], data['url'], 'field'],
+                    ['Username', data['username'], 'field'],
+                    ['Password', data['password'], 'field'],
+                    [data['dc_locator'], data['datacenter'], 'special select']
+                ],
+                orgs=[self.default_org],
+                org_select=False,
+                locations=[self.default_loc],
+                loc_select=True
+            )
+            # nasty hack
+            time.sleep(5)
+            search = self.compute_resource.search(data['name'])
+            self.assertIsNotNone(search)
+            self.compute_resource.delete(data['name'])
+            search = self.compute_resource.search(data['name'])
+            self.assertIsNone(search)
+
+    @data(
+        {'name': vmware_name,
+         'new_name': '%s-updated' % vmware_name,
+         'provider': FOREMAN_PROVIDERS['vmware'],
+         'url': vmware_vcenter,
+         'username': vmware_username,
+         'password': vmware_password,
+         'datacenter': vmware_datacenter,
+         'url_name': vmware_url_name,
+         'dc_locator': vmware_dc_locator}
+    )
+    def test_edit_compute_resource(self, data):
+        """ @Test: Edit a compute resource.
+        @Assert: A compute resource with new name exists
+        @Feature: Compute Resource
+        """
+        with Session(self.browser) as session:
+            make_resource(
+                session,
+                name=data['name'],
+                provider_type=data['provider'],
+                parameter_list=[
+                    [data['url_name'], data['url'], 'field'],
+                    ['Username', data['username'], 'field'],
+                    ['Password', data['password'], 'field'],
+                    [data['dc_locator'], data['datacenter'], 'special select']
+                ],
+                orgs=[self.default_org],
+                org_select=False,
+                locations=[self.default_loc],
+                loc_select=True
+            )
+            # nasty hack
+            time.sleep(5)
+            search = self.compute_resource.search(data['name'])
+            self.assertIsNotNone(search)
+
+            self.compute_resource.update(name=data['name'],
+                                         newname=data['new_name'])
+            search = self.compute_resource.search(data['new_name'])
+            self.assertIsNotNone(search)
+
+            self.compute_resource.delete(data['new_name'])
+            search = self.compute_resource.search(data['new_name'])
```

Comment:
self.compute_resource.delete(data['new_name'])->delete_entity is already have validation inside. You don't need:

search = self.compute_resource.search(data['new_name'])
self.assertIsNone(search)